formats/dot/internal/errors/errors.go

```diff
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// generated by gocc; DO NOT EDIT.
+// Code generated by gocc; DO NOT EDIT.
```

Comment:
> I think the copyright header does need to be on generated code. The generated code is the artifact, the process to generate it may be punching keys, or getting a machine to pretend to punch keys. The path to the product is not really relevant.

Right.

> I guess the best approach is to add some code to the make scripts that will replace the warning with the warning suffixed with the copyright notice.

Would you be willing to take a look at writing such a script? I'm not sure entirely how to do it in a clean manner, and where it makes sense to hook it in.

Should it be something written in Go with go:generate, or perhaps a shell script (e.g. `sed`) with regular expressions?

> Sorry for all this rigmarole.

That's ok. I am very glad that you have done a thorough review and I've vastly appreciated the feedback! Thanks.

Truth to be told, I had hoped the generated code would not need a copyright notice, seeing as we did not really write it as such. But that's fine. Once this last remark is addressed we should be closer to merging the PR and get decoding support into gonum/graph, which really excites me.

Have a lovely day

Cheers /u